go/client/logclient.go

```diff
@@ -105,6 +101,19 @@ type getEntryAndProofResponse struct {
 	AuditPath []string `json:"audit_path"` // the corresponding proof
 }
 
+type getProofByHashResponse struct {
+	LeafIndex    int      `json:"leaf_index"` // The 0-based index of the end entity corresponding to the "hash" parameter.
```

Comment:
We're using `int64` for leaf indices in general, is there a problem with the JSON parser that prevents using that?